CMakeLists.txt

```diff
@@ -99,15 +99,31 @@ set(CMAKE_INSTALL_PREFIX ${PROJECT_SOURCE_DIR})
 
 # Additional Fortran compiler flags.
 #
-# -fno-automatic: this was set in the original make_environment_gfortran_UBC file.
+#   -fno-automatic: this was set in the original make_environment_gfortran_UBC file.
 #
-# Note: optimization should be enabled on the Release target automatically.
+# Note: there is no need to pass optimization flags explicitly (e.g. -O2) -- CMake enables
+# -O2 automatically on the Release target. Similarly, -g is automatically set for the Debug
+# target.
 #
-# If need be, you can also set up linker flags. E.g.:
+# To set additional compiler flags, you can set CMAKE_Fortran_FLAGS when invoking CMake.
+# For example
 #
-#     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -static-libgfortran")
+#   cmake -DCMAKE_Fortran_FLAGS="-fno-automatic -fconvert=big-endian" ..
 #
-set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fno-automatic")
+# Note: this will override the defaults, so you need to pass -fno-automatic yourself. This
+# behavior is useful when using e.g. the Intel compiler, which has a different name for this
+# option, and so you don't want to pass a For GFortran-specific flag.
+#
+# Similarly, additional linker options can be passed via CMAKE_EXE_LINKER_FLAGS, e.g.
+#
+#   cmake -DCMAKE_EXE_LINKER_FLAGS="-static-libgfortran" ..
+#
+if(NOT DEFINED CMAKE_Fortran_FLAGS)
```

Comment:
Ok, this check does not actually work as intended. I think the `CMAKE_Fortran_FLAGS` variable is always defined, just set to an empty string or something like that. So this requires a different logic, but let's first figure out if we even need `-fno-automatic`.

Reply:
I'm pretty sure that flag is (or at least was) needed, but don't remember on top of my head why.